lib/LazyLoadingGhostFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager;
+
+use ProxyManager\Configuration;
+use ProxyManager\Factory\LazyLoadingGhostFactory as BaseLazyLoadingGhostFactory;
+
+/**
+ * Extends the LazyLoadingGhostFactory of the ProxyManager, to make it configureable according to the needs of Sulu.
+ */
+class LazyLoadingGhostFactory extends BaseLazyLoadingGhostFactory
+{
+    public function __construct($cacheDirectory)
+    {
+        if (!is_dir($cacheDirectory)) {
+            mkdir($cacheDirectory, 0777, true);
```

Comment:
`777` is the issue here: http://www.doctrine-project.org/2015/08/31/security_misconfiguration_vulnerability_in_various_doctrine_projects.html but it should probably be fixed globally in another PR.

Also, why not remove this class and just configure the `Configuration` in the DI container and set the cache directory there? The cache directory would typically already exist (e.g. `app/cache`).